### PR TITLE
Refactor SingleFileInput

### PR DESCRIPTION
**Pull request by jeremyyap**
- Extract image and badge previews into separate components for use with SingleFileInput
- Add a new default file preview to SingleFileInput that shows just the uploaded file name
- Add `accept="image/*"` to achievement form image uploader

### CODE REVIEW

File: client/app/lib/components/redux-form/SingleFileInput/index.jsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  fileLabelError: {
+    color: red500,
+    display: 'inline-block',
+  },
+};
+
+/**
+ * Creates a Single file input component for use with Redux Forms.
+ * Available in two display modes: circular Avatar/Badge display (default) and non-circular image display.
+ * To use non-circular image display, pass in the prop `isNotBadge` with boolean value `true`.
```
Comment (weiqingtoh) [inline client/app/lib/components/redux-form/SingleFileInput/index.jsx]: Suggest to update the documentation here.

File: client/app/lib/components/redux-form/SingleFileInput/index.jsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      name: PropTypes.string,
+    }),
+    input: PropTypes.shape({
+      onChange: PropTypes.func.isRequired,
+    }),
+    errorMessage: PropTypes.string,
+    required: PropTypes.bool,
+    intl: intlShape.isRequired,
+    accept: PropTypes.string,
+    previewComponent: PropTypes.func,
```
Comment (weiqingtoh) [inline client/app/lib/components/redux-form/SingleFileInput/index.jsx]: > The propTypes typechecking happens after defaultProps are resolved, so typechecking will also apply to the defaultProps

Shall we have this as `required`? Read this off the [React docs](https://reactjs.org/docs/typechecking-with-proptypes.html).

Comment (jeremyyap) [reply client/app/lib/components/redux-form/SingleFileInput/index.jsx]: We have a [lint rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/default-props-match-prop-types.md) in effect that requires the prop _not_ to be declared as required.

Comment (weiqingtoh) [reply client/app/lib/components/redux-form/SingleFileInput/index.jsx]: >_< ok ignore this!

**Pull request merged**